src/page/MyStorePage/components/CatagoryMenuList/index.tsx

```diff
@@ -1,41 +1,50 @@
 import styles from './CatagoryMenuList.module.scss';
 
-interface Props {
-  name: string;
-  menus: Menu[];
-}
-export default function CatagoryMenuList({ menus, name }: Props) {
+export default function CatagoryMenuList({ menus }: { menus: MenuCategory }) {
   return (
     <div>
       <div className={styles.category__title}>
-        {name}
-        {name === '이벤트 메뉴' ? (
+        {menus.name}
+        {menus.name === '이벤트 메뉴' ? (
           <EventMarkIcon className={styles['category__event-mark']} />) : null}
       </div>
-      <div className={styles.menu__wrapper}>
-        {menus.map((menu) => (menu.img === null ? (
-          <div key={menu.img} className={styles.menu__item}>
-            <div className={styles['menu__empty-img']}>
-              <CUTLERY className={styles['menu__empty-img-icon']} />
-              <span className={styles['menu__empty-img-caption']}>이미지 준비 중입니다.</span>
-            </div>
-            <div>
-              <span className={styles.menu__name} key={name}>{menu.name}</span>
-              <div className={styles.menu__price}>
-                {menu.option_price.map((option) => (
-                  <span className={styles['menu__price-unit']} key={option.option}>
-                    {option.price !== null
-                      ? `${option.option}:${option.price.toLocaleString()}` : `${option.option}`}
-                  </span>
-                ))}
+      <div className={styles['menu__slide-wrapper']}>
+        <div className={styles.menu__content}>
+          {menus.menus.map((menu) => (
+            <div key={menu.name} className={styles.menu__item}>
+              {menu.image_urls?.length === 0 ? (
+                <div className={styles['menu__non-img-border']}>
+                  <div className={styles['menu__empty-img']}>
+                    <CUTLERY className={styles['menu__empty-img-icon']} />
+                    <span className={styles['menu__empty-img-caption']}>이미지 준비 중입니다.</span>
+                  </div>
+                </div>
+              ) : (
+                <div className={styles['menu__img-border']}>
+                  <img src={menu.image_urls[0]} alt="menu" className={styles['menu__menu-img']} />
+                </div>
+              )}
+              <div>
+                <span className={styles.menu__name} key={menu.name}>{menu.name}</span>
+                <div className={styles.menu__price}>
+                  {menu.option_prices && menu.option_prices.map((option) => (
+                    <span className={styles['menu__price-unit']} key={option.option}>
+                      {`${option.option}:${option.price.toLocaleString()}원`}
+                    </span>
+                  ))}
+                  {menu.single_price && (
+                    <span className={styles['menu__price-unit']}>
+                      {`${menu.single_price.toLocaleString()}원`}
+                    </span>
+                  )}
```

Comment:
`MultiPriceMenu`와 `SinglePriceMenu`의 Union Type이기 때문에 두번째 조건이 없어도 충분히 `SinglePriceMenu`로 추론된다는 점을 활용해보는건 어떨까요??
그냥 적어만두면 쉽게 이해되지 않을 수 있으니 적절한 주석을 포함하는 것도 좋아보이네요
```suggestion
                  {menu.option_prices ? menu.option_prices.map((option) => (
                    <span className={styles['menu__price-unit']} key={option.option}>
                      {`${option.option}:${option.price.toLocaleString()}원`}
                    </span>
                  )) : (
                    <span className={styles['menu__price-unit']}>
                      {`${menu.single_price.toLocaleString()}원`}
                    </span>
                  )}
```

Reply:
간단한 주석으로 옵션 가격과 단일 가격을 구분해 두었습니다.
좀 더 똑똑한 방법으로 Union Type을 사용할 수 있었네요!